docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,197 @@
+increase performances for ingesting or querying the logs. It is also possible
+to fine tune the deployment configuration in order to have more replicas and
+ideally the Log centralisation service supports horizontal POD autoscaling.
+
+*Highly Available*
+
+Log collection, storage and browsing services can be replicated in order to
+ensure the same level of High Availability Prometheus has i.e. can lose all
+replicas but one.
+
+*Log Querying*
+
+Get all logs for a given period, node(s), pod regex, limited list of predefined
+labels and free keywords text.
+Typical Zenko use case: collecting all logs across several components, related
+to a S3 uniq request.
+
+*Log statistics (nice to have)*
+
+The Log centralisation service also offer the ability to consume statistics
+about the logs like the number of occurrences of one type of log during a
+certain period of time.
+
+*Monitorable/Observable service (health, performances and alerts)*
+
+The Platform Administrator can monitor capacity usage, IOPS, latency and
+bandwidth of the Log centralisation service. He can also monitor the health of
+the service (i.e. if some active alerts exist). He can visualize active and
+past alerts impacting the service.
+
+*Performances (TBD)*
+
+Typical workloads can generate around 1000 logs per second per node.
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  k8s and metalk8s services error investigation.
+
+- As a Platform Administrator, I want to browse non container (kubelet,
+  containerd, salt-minion, cron) logs (from all servers) from a
+  unique endpoint, in order to ease System error investigation.
+
+- As a Storage Administrator, I want to browse all Solution instance containers
+  logs (from all servers) from a unique endpoint, in order to ease Solution
+  instance Error investigation.
+
+- As a Platform Administrator, I want to push all containers logs to an
+  external log centralisation system, In Order to archive it or aggregate it
+  with other application logs.
+
+
+(some other US extracted from Loki design doc)
+
+- After receiving an alert on my service and drilling into the query associated
+  with said alert, I want to quickly see the logs associated with the jobs
+  which produced those timeseries at the time of the alert.
+
+- After a pod or node disappears, I want to be able to retrieve logs from just
+  before it died, so I can diagnose why it died.
+
+- After discovering an ongoing issue with my service, I want to extract a
+  metric from some logs and combine it with my existing time series data.
+
+- I have a legacy job which does not expose metrics about errors - it only logs
+  them.  I want to build an alert based on the rate of occurrences of errors in
+  the log.
+
+
+Deployment & Configuration
+--------------------------
+
+The log centralisation service is scheduled on infra nodes.
```

Comment:
> on infra nodes

Not only you also have promtail (or fluentd, fluent-bit) to install on every nodes to send logs

Reply:
That's not mandatory, and the requirements may impose that we have the ability to only run ingestors on a subset of the cluster nodes.

Reply:
will specify log storage service to run on infra nodes

Reply:
> That's not mandatory, and the requirements may impose that we have the ability to only run ingestors on a subset of the cluster nodes.

Really ? Then you may have log centralization with only a part of all the logs ? For example you will have salt-minion, kubelet, ... logs of infra nodes and that it...

To me, we really want to have it everywhere, it's weird to have log centralization for some nodes but not for others ... it's not really centralized in this case.
To me, if you add "ability to run ingestors on a subset of the cluster nodes" then you want to do the same for node exporter (and to me both does not make sense)

Reply:
No you're right, I didn't mean to ingest only logs for a subset of Nodes. I don't know why, but I had the wrong idea ingestors could consume logs from other Nodes, but that's not how it works in `promtail` or `fluent-bit` (even for Pod logs, they read from Node-local logfiles).
What would be nice is to make sure in the design that storage + as much compute resources as possible are consumed on the infra nodes, and leave the bare minimum in the ingestors (ideally just forward the text streams?).